pkg/tsdb/cloudwatch/response_parser.go

```diff
@@ -195,7 +195,14 @@ func buildDataFrames(ctx context.Context, startTime time.Time, endTime time.Time
 
 		name := label
 		var labels data.Labels
-		if features.IsEnabled(ctx, features.FlagCloudWatchNewLabelParsing) {
+		if query.GetGetMetricDataAPIMode() == models.GMDApiModeSQLExpression {
+			metricQueryLabel := label
+			if hasStaticLabel {
```

Comment:
Do you have to do this? It looks like this query mode returns actually returns the static label as the label (unless I just didn't try the right query)

Reply:
Without this I was getting results where the label returned from CW includes the `GROUP BY` field appended to the static label.

<img width="599" alt="Screenshot 2024-04-09 at 8 15 56 AM" src="https://github.com/grafana/grafana/assets/19530599/b83a8627-d0e8-4f38-b0cb-cd9f4b52c2db">

Reply:
Removed this change, the static label will only be applied to the field name instead of the labels.